src/main/java/frc/team2412/robot/Subsystems/IndexerSubsystem.java

```diff
@@ -0,0 +1,257 @@
+	}
+
+	public boolean allSensorsOff() {
+		if (!back.get() && !backMid.get() && !mid.get() && !frontMid.get() && !front.get()) {
+			return true;
+		}
+
+		return false;
+		// check if all sensors are on
+	}
+
+	public void shoot() {
+		// allShiftDown();
+		indexMidMotor.set(1);
+		if (ballUnbalancedSide == IntakeDirection.FRONT) {
+			indexBackMotor.set(-1);
+		} else {
+			indexFrontMotor.set(1);
+		}
+		if (frontMid.get() && front.get()) {
+			indexFrontMotor.set(0);
+			ballUnbalancedSide = IntakeDirection.FRONT;
+		} else if (backMid.get() && back.get()) {
+			indexBackMotor.set(0);
+			ballUnbalancedSide = IntakeDirection.BACK;
+		} else if (allSensorsOff()) {
+			stopAll();
+			ballUnbalancedSide = IntakeDirection.NONE;
+		}
+	}
+
+//Switch ball positions
+
+	public enum IntakeDirection {
+		FRONT, BACK, BOTH, NONE;
+	}
+
+	public void intake(int num, IntakeDirection dir) {
+		switch (num) {
+		case 1:
+			if (dir == IntakeDirection.FRONT) {
+				indexFrontMotor.set(1);
+			} else if (dir == IntakeDirection.BACK) {
+				indexBackMotor.set(-1);
+			}
+			if (!mid.get()) {
+				stopAll();
+				numBalls++;
+			}
+			break;
+		case 2:
+			indexFrontMotor.set(1);
+			indexBackMotor.set(-1);
+			if (!frontMid.get()) {
+				stopAll();
+				ballUnbalancedSide = IntakeDirection.BACK;
+				numBalls++;
+			} else if (!backMid.get()) {
+				stopAll();
+				ballUnbalancedSide = IntakeDirection.FRONT;
+				numBalls++;
+			}
+			break;
+		case 3:
+			if (ballUnbalancedSide == IntakeDirection.FRONT) {
+				indexFrontMotor.set(1);
+				if (!frontMid.get()) {
+					stopAll();
+					numBalls++;
+					ballUnbalancedSide = IntakeDirection.NONE;
+
+				} else if (!intakeBack.get()) {
+					stopAll();
+					numBalls++;
+				}
+			} else if (ballUnbalancedSide == IntakeDirection.BACK) {
+				indexBackMotor.set(-1);
+				if (!backMid.get()) {
+					stopAll();
+					numBalls++;
+					ballUnbalancedSide = IntakeDirection.NONE;
+				} else if (!intakeFront.get()) {
+					stopAll();
+					numBalls++;
+				}
+
+			}
+			break;
+		case 4:
+
+			if (ballUnbalancedSide == IntakeDirection.FRONT) {
+				indexFrontMotor.set(1);
+				if (!frontMid.get()) {
+					stopAll();
+					numBalls++;
+					ballUnbalancedSide = IntakeDirection.NONE;
+
+				} else if (!intakeBack.get()) {
+					stopAll();
+					numBalls++;
+				}
+			} else if (ballUnbalancedSide == IntakeDirection.BACK) {
+				indexBackMotor.set(-1);
+				if (!backMid.get()) {
+					stopAll();
+					numBalls++;
+					ballUnbalancedSide = IntakeDirection.NONE;
+				} else if (!intakeFront.get()) {
+					stopAll();
+					numBalls++;
+				}
+
+			}
+			break;
+		case 5:
+			if (dir == IntakeDirection.FRONT) {
+				if (!intakeFront.get()) {
+					numBalls++;
+				}
+			} else if (dir == IntakeDirection.BACK) {
+				indexBackMotor.set(1);
+				if (!intakeBack.get()) {
+					numBalls++;
+				}
+			}
+			break;
+		}
+
+	}
+
+	public void swap(IntakeDirection dir, int num) {
+		// dir is side with least balls

```

Comment:
Why can't this simply read the sensors to see where the power cells currently are?  It seems fragile to depend on all other code paths to keep track of which power cells are where.  Since this code already depends on the sensors reporting the right thing for the end condition, it seems reasonable to depend on them for the start condition as well.